docs/implementations/containerd.md

```diff
@@ -0,0 +1,68 @@
+# Containerd
+Containerd is a container runtime.  This document describes how it authenticates to remote registries.
+## Authentication
+- Authentication information is stored on a per host basis.
+    - `service` and `realm` are assumed to be constant per host.
+    - Individual `scope` values are added without any sort of merging (i.e. stored as both `repository:<repo>:pull` and `repository:<repo>:pull,push` instead of just `repository:<repo>:pull,push`).
+    - This also applies to hosts that are the target of redirection.
+- Requests are retried up to 5 times. Only status codes 401 (Unauthorized), 405 (Method Not Allowed), 408 (Request Timeout), and 429 (Too Many Requests) are retried.
+- Multiple scopes are included in push or repository remounts
```

Comment:
Are these scopes generated by containerd, read from the WWW-Authenticate header, or both?

Reply:
They all come from the www-authenticate header.